File: Mudit Choraria/longest-continuous-increasing-subsequence.cpp
class Solution {
public:
    int findLengthOfLCIS(vector<int>& nums) {
        int count = 1, ans = INT_MIN;
        for(int i = 1; i < size(nums); i++) {
            if(nums[i - 1] < nums[i]) {
                // keep increasing the count until a decreasing pair is found
                count++;
            } else {
                // reset the count when a decreasing pair is found
                ans = max(ans, count);
                count = 1;
            }
        }
        return max(ans, count);
    }
};

File: Parth Mathur/Longest Continuous Increasing Subsequence.java
class Solution {
    public int findLengthOfLCIS(int[] nums) {
        int current_right = 0, best_right = 0;
        for(int i = 0; i < nums.length; i++) {
            if(i == 0 || nums[i - 1] >= nums[i]) {
                current_right = 1;
            } else {
                current_right++;
            }
            best_right = Math.max(best_right, current_right);
        }
        return best_right;
    }
}
